src/tls.c

```diff
@@ -5351,6 +5351,165 @@ int TLSX_EncryptThenMac_Respond(WOLFSSL* ssl)
 
 #endif /* HAVE_ENCRYPT_THEN_MAC && !WOLFSSL_AEAD_ONLY */
 
+
+#if defined(WOLFSSL_DTLS) && defined(WOLFSSL_SRTP)
+
+/******************************************************************************/
+/* DTLS SRTP (Secure Real-time Transport Protocol)                            */
+/******************************************************************************/
+
+/* Only support single SRTP profile */
+typedef struct TlsxSrtp {
+    word16 len;
+    word16 id;
+} TlsxSrtp;
+
+static int TLSX_UseSRTP_GetSize(TlsxSrtp *srtp)
+{
+    /* only supports single profile */
+    const int profileCount = 1;
+    (void)srtp;
+    /*   SRTP Profile Len (2)
+     *      SRTP Profiles (2)
+     *   MKI (master key id) Length */
+    return (OPAQUE16_LEN + (profileCount * OPAQUE16_LEN) + 1);
+}
+
+static TlsxSrtp* TLSX_UseSRTP_New(word16 id, void* heap)
+{
+    TlsxSrtp* srtp;
+
+    srtp = (TlsxSrtp*)XMALLOC(sizeof(TlsxSrtp), heap, DYNAMIC_TYPE_TLSX);
+    if (srtp == NULL) {
+        WOLFSSL_MSG("TLSX SRTP Memory failure");
+        return NULL;
+    }
+    srtp->len = 2;
+    srtp->id = id;
+
+    return srtp;
+}
+
+static void TLSX_UseSRTP_Free(TlsxSrtp *srtp, void* heap)
+{
+    if (srtp != NULL) {
+        XFREE(srtp, heap, DYNAMIC_TYPE_TLSX);
+    }
+    (void)heap;
+}
+
+static int TLSX_UseSRTP_Parse(WOLFSSL* ssl, const byte* input, word16 length,
+    byte isRequest)
+{
+    int ret = BAD_FUNC_ARG;
+#ifndef NO_WOLFSSL_SERVER
+    int i;
+    TlsxSrtp* srtp = NULL;
+    word16 profile_len = 0;
+    word16 profile_value = 0;
+    word16 offset = 0;
+#endif
+
+    if (length < OPAQUE16_LEN) {
+        return BUFFER_ERROR;
+    }
+
+    if (!isRequest) {
+#ifndef NO_WOLFSSL_CLIENT
+        ssl->dtlsSrtpProfile = ssl->ctx->dtlsSrtpProfile;
+        ret = 0; /* success */
+#endif
+    }
+#ifndef NO_WOLFSSL_SERVER
+    else {
+        /* total length, not include itself */
+        ato16(input, &profile_len);
+        offset += OPAQUE16_LEN;
+        /* parse remainder one profile at a time, looking for match in CTX */
+        for (i=offset; i<length; i+=OPAQUE16_LEN) {
+            ato16(input+i, &profile_value);
```

Comment:
are we accounting for the MKI field in this parsing?

Reply:
No, the MKI is a byte that comes at the end. Always zero on send and ignored on parse. None of the implementations I found used MKI, so I did not add support for it.